From: Rene Kita <mail@rkta.de>
Date: Sun Sep  1 11:02:34 CEST 2024
Subject: Add keybinding to insert current search term
Patch: patches/Add-keybinding-to-insert-current-search-term.patch

---
 linein.c |   13 ++++++++++---
 main.c   |    2 +-
 2 files changed, 11 insertions(+), 4 deletions(-)

--- a/linein.c
+++ b/linein.c
@@ -33,7 +33,7 @@ static char **CFileBuf = NULL;
 static int NCFileBuf;
 static int NCFileOffset;
 
-static void _iword(void), _cy(void), _paste(void),
+static void _isrch(void), _iword(void), _cy(void), _paste(void),
 _mvR(void), _mvL(void), _mvRw(void), _mvLw(void), delC(void), insC(void),
 _mvB(void), _mvE(void), _enter(void), _quo(void), _bs(void), _bsw(void),
 killn(void), killb(void), _inbrk(void), _esc(void), _editor(void),
@@ -58,7 +58,7 @@ void (*InputKeymap[32]) (void) = {
 /*  C-p     C-q     C-r     C-s     C-t     C-u     C-v     C-w     */
     _prev,  _quo,   _bsw,   _iword,  _mvLw,  killb,  _quo,   _bsw,
 /*  C-x     C-y     C-z     C-[     C-\     C-]     C-^     C-_     */
-    _tcompl,_cy,    _iword, _esc,   _iword, _iword, _iword, _iword,
+    _tcompl,_cy,    _iword, _esc,   _isrch, _iword, _iword, _iword,
 };
 /* *INDENT-ON* */
 
@@ -211,7 +211,7 @@ inputLineHistSearch(const char *prompt,
 	    if (incrfunc == NULL
 		|| (c = incrfunc((int)c, strBuf, strProp)) < 0x20)
 		(*InputKeymap[(int)c]) ();
-	    if (c > 0x20) {
+	    if (c != 255 && c >= 0x20) {
 		tmp = wc_char_conv(c);
 		ins_char(tmp);
 	    }
@@ -623,6 +623,13 @@ _iword(void)
     ins_char(Strnew_charp(GetWord(Currentbuf)));
 }
 
+extern const char *SearchString;
+static void
+_isrch(void)
+{
+    ins_char(Strnew_charp(SearchString));
+}
+
 static void
 _quo(void)
 {
--- a/main.c
+++ b/main.c
@@ -86,7 +86,7 @@ static void SigPipe(SIGNAL_ARG);
 #ifdef USE_MARK
 static const char *MarkString = NULL;
 #endif
-static const char *SearchString = NULL;
+const char *SearchString = NULL;
 int (*searchRoutine) (Buffer *, const char *);
 
 #ifndef __MINGW32_VERSION
